[k8s.io] ConfigMap should be consumable via environment variable [Conformance] {E2eNode Suite}

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet/2498/
Failed: [k8s.io] ConfigMap should be consumable via environment variable [Conformance] {E2eNode Suite}

```
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/configmap.go:382
Expected error:
    <*errors.errorString | 0xc4202685a0>: {
        s: "failed to get logs from pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019 for env-test: the server could not find the requested resource (get pods pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019)",
    }
    failed to get logs from pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019 for env-test: the server could not find the requested resource (get pods pod-configmaps-59ebd4af-087b-11e7-90c9-42010a800019)
not to have occurred
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2195
```


Previous issues for this test: #42526

This appears to have found a segfault:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xddd7fb]
goroutine 71645 [running]:
panic(0x2d90800, 0xc42000e040)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/kubernetes/pkg/kubelet/cm.(*cgroupManagerImpl).Pids.func1(0xc420d33860, 0x9d, 0x0, 0x0, 0x4d247e0, 0xc421d0aa50, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/cgroup_manager_linux.go:452 +0x3b
path/filepath.walk(0xc420d334a0, 0x97, 0x4d48b00, 0xc421ba5e10, 0xc421157880, 0x0, 0x0)
        /usr/local/go/src/path/filepath/path.go:372 +0x22e
path/filepath.walk(0xc4221f73e0, 0x56, 0x4d48b00, 0xc421ba5d40, 0xc421157880, 0x0, 0x1)
        /usr/local/go/src/path/filepath/path.go:376 +0x344
path/filepath.Walk(0xc4221f73e0, 0x56, 0xc421157880, 0x0, 0x0)
        /usr/local/go/src/path/filepath/path.go:398 +0xd5
k8s.io/kubernetes/pkg/kubelet/cm.(*cgroupManagerImpl).Pids(0xc42038ab20, 0xc421b19440, 0x3c, 0xc42108b728, 0xc4223dafc0, 0x1)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/cgroup_manager_linux.go:466 +0x380
k8s.io/kubernetes/pkg/kubelet/cm.(*podContainerManagerImpl).tryKillingCgroupProcesses(0xc421f08960, 0xc421b19440, 0x3c, 0x43d515, 0xc42108b6d8)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/pod_container_manager_linux.go:120 +0x7e
k8s.io/kubernetes/pkg/kubelet/cm.(*podContainerManagerImpl).Destroy(0xc421f08960, 0xc421b19440, 0x3c, 0x1, 0xc421fad540)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/pod_container_manager_linux.go:157 +0x5a
created by k8s.io/kubernetes/pkg/kubelet.(*Kubelet).cleanupOrphanedPodCgroups
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_pods.go:1554 +0x381
```

cc @kubernetes/sig-node-bugs @derekwaynecarr @sjenning @dchen1107 @yujuhong @vishh

--- a/pkg/kubelet/cm/cgroup_manager_linux.go
+++ b/pkg/kubelet/cm/cgroup_manager_linux.go
@@ -449,12 +449,16 @@ func (m *cgroupManagerImpl) Pids(name CgroupName) []int {
 
 		// WalkFunc which is called for each file and directory in the pod cgroup dir
 		visitor := func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				glog.V(4).Infof("cgroup manager encountered error scanning cgroup path %q: %v", path, err)
+				return filepath.SkipDir
+			}
 			if !info.IsDir() {
 				return nil
 			}
 			pids, err = getCgroupProcs(path)
 			if err != nil {
-				glog.V(5).Infof("cgroup manager encountered error getting procs for cgroup path %v", path)
+				glog.V(4).Infof("cgroup manager encountered error getting procs for cgroup path %q: %v", path, err)
 				return filepath.SkipDir
 			}
 			pidsToKill.Insert(pids...)
@@ -464,7 +468,7 @@ func (m *cgroupManagerImpl) Pids(name CgroupName) []int {
 		// container cgroups haven't been GCed yet. Get attached processes to
 		// all such unwanted containers under the pod cgroup
 		if err = filepath.Walk(dir, visitor); err != nil {
-			glog.V(5).Infof("cgroup manager encountered error scanning pids for directory: %v", dir)
+			glog.V(4).Infof("cgroup manager encountered error scanning pids for directory: %q: %v", dir, err)
 		}
 	}
 	return pidsToKill.List()

